### PR TITLE
Add initial binary package for a Sway session

### PR DESCRIPTION
**Pull request by kgilmer**
Here is a sub for a new Sway session.  This represents the top level depencency for the session.  As you may notice when comparing to the flashback session, at this point in the package graph there is very little regolith packages referenced.  So there is a lot of flexibility.

### CODE REVIEW

**Comment by SoumyaRanjanPatnaik** [comment]
Ready to merge

**Review by SoumyaRanjanPatnaik — APPROVED** [review]

File: debian/control [hunk 1]
```diff
@@ -30,3 +30,22 @@ Description: Regolith customized gnome-flashback session
  This package contains the desktop and xsession configuration
  necessary to start a Regolith gnome-flashback / i3 desktop session.
  It is based on https://github.com/deuill/i3-gnome-flashback.
+
+Package: regolith-session-sway
+Architecture: any
+Depends: ${misc:Depends},
+    dbus # add more dependencies here,
```
Comment (kgilmer) [inline debian/control]: correctness: This line will fail to parse, can you remove the comment?

File: debian/control [hunk 2]
```diff
@@ -30,3 +30,22 @@ Description: Regolith customized gnome-flashback session
...
+    sway,
+    swayidle,
+    xwayland,
+    clipman,
+    wl-clipboard,
+    #trawld,
```
Comment (kgilmer) [inline debian/control]: correctness: I believe this will fail to parse because due to the comment, there is nothing after `wl-clipboard` to read.  Either remove the comma on line 42 or enable the commented out lines.

File: usr/bin/regolith-diagnostic [hunk 3]
```diff
@@ -47,3 +47,6 @@ gsettings list-recursively org.regolith-linux.ilia
 gsettings list-recursively org.gnome.desktop.wm.keybindings
 gsettings list-recursively org.gnome.settings-daemon.plugins.media-keys
 gsettings list-recursively org.gnome.desktop.wm.preferences
+
+echo "--- trawl"
+trawl -q
```
Comment (kgilmer) [inline usr/bin/regolith-diagnostic:52]: nice!

File: usr/bin/regolith-session-init [hunk 4]
```diff
@@ -5,28 +5,47 @@ set -Eeu -o pipefail
...
-
-load_standard_xres  # ~/.Xresources
-load_regolith_xres  # ~/.config/regolith2/Xresources
-
-# Register with gnome-session so that it does not kill the whole session thinking it is dead.
-# See https://zork.net/~st/jottings/gnome-i3.html for details
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.RegisterClient "string:Regolith" "string:$DESKTOP_AUTOSTART_ID"
-fi
+start_i3 () {
```
Comment (kgilmer) [inline usr/bin/regolith-session-init]: I'm fine to merge this as is but wonder if you considered breaking out the wayland specific login into a separate file.  IMO this would be a better approach as it scales better.  EG if I make a change to this file as-is I may have to test both x11 and sway, whereas if they were discrete files I wouldn't.  Is there a reason why breaking the logic out is problematic?

File: usr/bin/regolith-session-init [hunk 5]
```diff
@@ -5,28 +5,47 @@ set -Eeu -o pipefail
...
-if [ -f "$USER_POST_LOGOUT_SCRIPT_FILE" ] ; then
-    echo "Regolith is launching user post-logout file $USER_POST_LOGOUT_SCRIPT_FILE"
-	timeout --verbose 5 bash "$USER_POST_LOGOUT_SCRIPT_FILE"
-fi
-
-# Close session when i3 exits.
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.Logout "uint32:1"
+export MOZ_ENABLE_WAYLAND=1
+export STUDIO_JDK=/usr/lib/jvm/java-11-openjdk/
```
Comment (kgilmer) [inline usr/bin/regolith-session-init]: Hmm..what is this?

File: usr/bin/regolith-session-init [hunk 6]
```diff
@@ -5,28 +5,47 @@ set -Eeu -o pipefail
...
-# Run user's post logout script if script. Post logout means that the X session is closed at this point.
-if [ -f "$USER_POST_LOGOUT_SCRIPT_FILE" ] ; then
-    echo "Regolith is launching user post-logout file $USER_POST_LOGOUT_SCRIPT_FILE"
-	timeout --verbose 5 bash "$USER_POST_LOGOUT_SCRIPT_FILE"
-fi
-
-# Close session when i3 exits.
-if [ -n "$DESKTOP_AUTOSTART_ID" ]; then
-    dbus-send --print-reply --session --dest=org.gnome.SessionManager "/org/gnome/SessionManager" org.gnome.SessionManager.Logout "uint32:1"
+export MOZ_ENABLE_WAYLAND=1
```
Comment (kgilmer) [inline usr/bin/regolith-session-init]: Why is this needed?

File: usr/share/applications/regolith.desktop [hunk 7]
```diff
@@ -2,7 +2,7 @@
 Type=Application
 Name=regolith
 Comment=Productivity-focused desktop environment, Regolith
-Exec=regolith-session-init
+Exec=regolith-session-init i3
```
Comment (kgilmer) [inline usr/share/applications/regolith.desktop]: interesting strategy.  I would there be a problem if we simply execute separate init scrips for each backend rather than having the script handle both?

File: usr/lib/regolith/regolith-session-common.sh [hunk 8]
```diff
@@ -65,3 +84,22 @@ load_regolith_xres() {
...
 }
+
+# Loads the Xresource files into trawl. 
+# NOTE: Will load sway specific reource files
+load_regolith_trawlres() {
+    if [ -f "$USER_XRESOURCE_OVERRIDE_FILE" ]; then
+        trawldb --merge "$USER_XRESOURCE_OVERRIDE_FILE" 
+    fi
+
+    LOOK_STYLE_ROOT_PATH=$(trawlcat regolith.look.path "$DEFAULT_XRESOURCE_LOOK_PATH")
```
Comment (kgilmer) [inline usr/lib/regolith/regolith-session-common.sh:106]: nice!

**Review by kgilmer — COMMENTED** [review]
A few questions and fixes needed to debian packaging data.  Great work!

File: debian/control (hunk 2 above)
Comment (SoumyaRanjanPatnaik) [reply debian/control]: Actually I intend to have those lines enabled. But for now these packages don't really exist😅. I'll probably remove these for now and add them back once everything is packaged

File: usr/bin/regolith-session-init (hunk 4 above)
Comment (SoumyaRanjanPatnaik) [reply usr/bin/regolith-session-init]: I thought it would be convenient to have the same starter script for both sway and i3. But on second thought, what you're saying does make a lot of sense.

File: usr/share/applications/regolith.desktop (hunk 7 above)
Comment (SoumyaRanjanPatnaik) [reply usr/share/applications/regolith.desktop]: Actually no. Having separate scripts would work just the same.

File: usr/bin/regolith-session-init (hunk 5 above)
Comment (SoumyaRanjanPatnaik) [reply usr/bin/regolith-session-init]: Some applications require additional environment variables to work well with Wayland. It's quite common to have such variable definitions included in starter scripts. And these variables need to be declared before sway starts.

File: usr/bin/regolith-session-init (hunk 6 above)
Comment (SoumyaRanjanPatnaik) [reply usr/bin/regolith-session-init]: In my experience, without this Firefox will use legacy playback methods (as in hardware acceleration won't work). This leads to unacceptable CPU usage and occasional jittering when playing in 4k.

Comment (SoumyaRanjanPatnaik) [reply usr/bin/regolith-session-init]: There's also the added benefit of support for touchpad gestures (like pinch to zoom) and maybe a few other features that can only work in Wayland.

File: usr/bin/regolith-session-init (hunk 5 above)
Comment (SoumyaRanjanPatnaik) [reply usr/bin/regolith-session-init]: The second line is necessary for a lot of Java based applications (especially for Intellij based IDEs) to work properly.

Comment (SoumyaRanjanPatnaik) [reply usr/bin/regolith-session-init]: Although, I think it might be a good idea to let users define their pre-startup environment variables at one place. What are your thoughts on this?

**Comment by SoumyaRanjanPatnaik** [comment]
I've made some changes based on your comment. Can you review them?

File: usr/bin/regolith-session-init (hunk 5 above)
Comment (kgilmer) [reply usr/bin/regolith-session-init]: My opinion is app specific logic should not live in general-purpose scripts.  For a user that does not use Firefox (assuming that from `Moz`) or care about java programs would classify this as bloat IMO.  Also, what's to say that the JVM path is the one a user really wants Inteliij to point to, or is even valid?

The simplest thing we can do is provide guidance/docs for poorly behaving apps that users encounter.  More work would be to create packages that install support for some applications like these.  I'm guessing there is something in Wayland like `/etc/X11/Xsession` that allows the user environment to be extended by a package.

File: usr/bin/regolith-session-init (hunk 6 above)
Comment (kgilmer) [reply usr/bin/regolith-session-init]: As mentioned, I think this belongs elsewhere or possibly just in documentation.

File: etc/regolith/sway/environment [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# For using firefox with wayland
```
Comment (kgilmer) [inline etc/regolith/sway/environment:1]: Kindly remove these for now

File: usr/bin/regolith-session-sway [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+set -Eeu -o pipefail
+
+source /usr/lib/regolith/regolith-session-common.sh
+
+# --------- Script Main -----------
+resolve_default_config_file
+
+# Preloads user environment variables
+USER_ENVIRONMENT=${trawlcat sway.preload_env $HOME/.config/regolith2/sway/environment}
```
Comment (kgilmer) [inline usr/bin/regolith-session-sway]: I would prefer if this could be expressed in a standard way if possible.

https://wiki.gnome.org/Initiatives/Wayland/SessionStart

https://github.com/sddm/sddm/issues/916

https://superuser.com/questions/992210/how-to-set-environment-variables-for-a-gnome-wayland-session

Hmm, seems like maybe Wayland doesn't provide anything (yet)..?  Are you aware of anything that my 5 mins of google searching didn't uncover?  If not let's keep this logic but remove the default that has the moz and jdk stuff.

File: usr/bin/regolith-session-sway [hunk 11]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# This script initializes Regolith Xresources and launches i3.
```
Comment (kgilmer) [inline usr/bin/regolith-session-sway:2]: copypasta

File: etc/regolith/sway/environment (hunk 9 above)
Comment (kgilmer) [reply etc/regolith/sway/environment:1]: I'd like to begin testing this session from a built package so we'll address these app specific configs later.
